pyensign/ensign.py

```diff
@@ -77,7 +77,7 @@ def __init__(
         else:
             self.topics = Cache()
 
-    async def publish(self, topic_name, *events, client_id=""):
+    async def publish(self, topic_name, event_source, client_id=""):
```

Comment:
Thank you for figuring out how to support generators passed in from the user. I would like to keep the `*events` syntax, since that means the user can do either of the following.

```python
await client.publish(topic_name, event)
await client.publish(topic_name, event1, event2)
await client.publish(topic_name, event_gen)
```

So we may just need a way to determine if the input is an `event_source` or events themselves. Can we use a check something like this?

```python
if inspect.isgeneratorfunction(event):
    # Use the provided generator
else:
    # Create the generator like before
```

Reply:
Hi @pdeziel , yes we can surely determine if input is generator or not. But if use the existing syntax `*events` wont it result in the method having a tuple of generator object rather than the generator object user provided, if I am not wrong.

Reply:
I think that's correct, so maybe in that case what we need to do is iterate over the `*events`, and for each object in the tuple if it's a generator we would call `next()` on it until it's completely consumed, otherwise just assume it's a single event?

Reply:
Yes, that's what I thought of. It indirectly enables the user to provide event(s), generator(s) or a combination of both, so we'll have to iterate over each object and identify if its an event or a generator and then process accordingly. 
Not a clean solution though, a better solution would be to seperate events and generators. What do you suggest?

Reply:
Ok, another option is to add another user method which just takes a generator, but let's give it a name like `publish_from(topic_name, event_generator)` to distinguish it from `publish(topic_name, *events)`. If we need to reduce repeated code, we can add an internal method `_publish()` which both methods call that just accepts a generator and calls the client to actually do the publish.

It's more in line with what you originally suggested too, does that make sense?

Reply:
Yes. Sounds good. I'll make these changes